tc/core/tensor-inl.h

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <algorithm>
+#include <sstream>
+
+#include <glog/logging.h>
+
+namespace tc {
+
+template <typename DLTensorType>
+inline void setSizes(DLTensorType& t, const std::vector<int64_t>& sizes) {
+  auto ndim = sizes.size();
+  for (size_t i = 0; i < ndim; ++i) {
+    t.shape[i] = sizes[i];
+  }
+}
+
+template <typename DLTensorType>
+inline void setStrides(DLTensorType& t, const std::vector<int64_t>& strides) {
+  auto ndim = strides.size();
+  for (size_t i = 0; i < ndim; ++i) {
+    t.strides[i] = strides[i];
+  }
+}
+
+template <typename T>
+std::vector<T> makeStridesFromSizes(const std::vector<T>& sizes) {
+  auto ndim = sizes.size();
+  if (ndim == 0) {
+    return std::vector<T>();
+  }
+  std::vector<int64_t> strides(sizes.size(), 0);
+  strides[ndim - 1] = 1;
+  for (int i = ndim - 2; i >= 0; --i) {
+    strides[i] = strides[i + 1] * sizes[i + 1];
+  }
+  return strides;
+}
+
+inline DLContext getCPUDLContext() {
+  DLContext res;
+  res.device_id = 0;
+  res.device_type = DLDeviceType::kDLCPU;
+  return res;
+}
+
+inline DLContext getGPUDLContext(int device_id) {
+  DLContext res;
+  res.device_id = device_id;
+  res.device_type = DLDeviceType::kDLGPU;
+  return res;
+}
+
+template <typename T>
+DLDataType getDLDataType();
+
+template <>
+inline DLDataType getDLDataType<float>() {
```

Comment:
Don't we want other types (`double`, other `(u)int` types) ? 
I don't see this function used though.  Add a TODO maybe...

Reply:
well dead code should die, done